From 9f88af04f03d585b8257740745d19897b48a9795 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Fri, 18 Jun 2021 10:41:42 +0900
Subject: [PATCH] ksmbd: remove ____ksmbd_align in ksmbd_server.h
Git-commit: 9f88af04f03d585b8257740745d19897b48a9795
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


None of structures needs the attribute.
So remove ____ksmbd_align in ksmbd_server.h.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/ksmbd_server.h | 38 +++++++++++++++++---------------------
 1 file changed, 17 insertions(+), 21 deletions(-)

diff --git a/fs/cifsd/ksmbd_server.h b/fs/cifsd/ksmbd_server.h
index c2467a709144..a915ca5596dc 100644
--- a/fs/cifsd/ksmbd_server.h
+++ b/fs/cifsd/ksmbd_server.h
@@ -13,10 +13,6 @@
 #define KSMBD_GENL_NAME		"SMBD_GENL"
 #define KSMBD_GENL_VERSION		0x01
 
-#ifndef ____ksmbd_align
-#define ____ksmbd_align		__aligned(4)
-#endif
-
 #define KSMBD_REQ_MAX_ACCOUNT_NAME_SZ	48
 #define KSMBD_REQ_MAX_HASH_SZ		18
 #define KSMBD_REQ_MAX_SHARE_NAME	64
@@ -51,19 +47,19 @@ struct ksmbd_startup_request {
 	__u32	share_fake_fscaps;
 	__u32	sub_auth[3];
 	__u32	ifc_list_sz;
-	__s8	____payload[0];
-} ____ksmbd_align;
+	__s8	____payload[];
+};
 
 #define KSMBD_STARTUP_CONFIG_INTERFACES(s)	((s)->____payload)
 
 struct ksmbd_shutdown_request {
 	__s32	reserved;
-} ____ksmbd_align;
+};
 
 struct ksmbd_login_request {
 	__u32	handle;
 	__s8	account[KSMBD_REQ_MAX_ACCOUNT_NAME_SZ];
-} ____ksmbd_align;
+};
 
 struct ksmbd_login_response {
 	__u32	handle;
@@ -73,12 +69,12 @@ struct ksmbd_login_response {
 	__u16	status;
 	__u16	hash_sz;
 	__s8	hash[KSMBD_REQ_MAX_HASH_SZ];
-} ____ksmbd_align;
+};
 
 struct ksmbd_share_config_request {
 	__u32	handle;
 	__s8	share_name[KSMBD_REQ_MAX_SHARE_NAME];
-} ____ksmbd_align;
+};
 
 struct ksmbd_share_config_response {
 	__u32	handle;
@@ -90,8 +86,8 @@ struct ksmbd_share_config_response {
 	__u16	force_uid;
 	__u16	force_gid;
 	__u32	veto_list_sz;
-	__s8	____payload[0];
-} ____ksmbd_align;
+	__s8	____payload[];
+};
 
 #define KSMBD_SHARE_CONFIG_VETO_LIST(s)	((s)->____payload)
 #define KSMBD_SHARE_CONFIG_PATH(s)				\
@@ -111,43 +107,43 @@ struct ksmbd_tree_connect_request {
 	__s8	account[KSMBD_REQ_MAX_ACCOUNT_NAME_SZ];
 	__s8	share[KSMBD_REQ_MAX_SHARE_NAME];
 	__s8	peer_addr[64];
-} ____ksmbd_align;
+};
 
 struct ksmbd_tree_connect_response {
 	__u32	handle;
 	__u16	status;
 	__u16	connection_flags;
-} ____ksmbd_align;
+};
 
 struct ksmbd_tree_disconnect_request {
 	__u64	session_id;
 	__u64	connect_id;
-} ____ksmbd_align;
+};
 
 struct ksmbd_logout_request {
 	__s8	account[KSMBD_REQ_MAX_ACCOUNT_NAME_SZ];
-} ____ksmbd_align;
+};
 
 struct ksmbd_rpc_command {
 	__u32	handle;
 	__u32	flags;
 	__u32	payload_sz;
-	__u8	payload[0];
-} ____ksmbd_align;
+	__u8	payload[];
+};
 
 struct ksmbd_spnego_authen_request {
 	__u32	handle;
 	__u16	spnego_blob_len;
 	__u8	spnego_blob[0];
-} ____ksmbd_align;
+};
 
 struct ksmbd_spnego_authen_response {
 	__u32	handle;
 	struct ksmbd_login_response	login_response;
 	__u16	session_key_len;
 	__u16	spnego_blob_len;
-	__u8	payload[0];		/* session key + AP_REP */
-} ____ksmbd_align;
+	__u8	payload[];		/* session key + AP_REP */
+};
 
 /*
  * This also used as NETLINK attribute type value.
-- 
2.33.0

